lms/static/scripts/frontend_apps/config.ts

```diff
@@ -249,6 +250,30 @@ const defaultConfig = { api: {} } as unknown;
 
 export const Config = createContext(defaultConfig as ConfigObject);
 
+/**
+ * Utility that modifies a type by making properties matching `K` non-nullable.
+ */
+export type Ensure<T, K extends keyof T> = T & Required<Pick<T, K>>;
+
+/**
+ * Return the backend-provided configuration, requiring that the optional
+ * fields specified by `requiredKeys` are set.
+ *
+ * nb. The `= never` makes the return type just `ConfigObject` if `requiredKeys`
+ * is omitted, otherwise it would change all optional keys to required.
+ */
+export function useConfig<R extends keyof ConfigObject = never>(
+  requiredKeys: R[] = []
+): Ensure<ConfigObject, R> {
+  const config = useContext(Config);
+  for (const key of requiredKeys) {
+    if (!(key in config)) {
+      throw new Error(`Required configuration key "${key}" not set`);
+    }
```

Comment:
I guess it's mostly an edge case, but this condition won't trigger if the key is actually there but it has value `undefined`.

![image](https://user-images.githubusercontent.com/2719332/217773659-49c9ca38-7146-4108-963e-c1817690c890.png)

Should we maybe check `!config[key]` instead?

Reply:
Settings might legitimately have falsey values (eg. `false` for a disabled flag or `0` for a zero count). I chose to define "field is set" to mean that the object has a property here, though I think comparison to `undefined` would work just as well.  We _might_ find that it will be more convenient to treat all nullish values as "not set" in future. That would allow Python code to do something like `config["foo"] = some_value if condition else None`.